nncf/experimental/tensor/functions.py

```diff
@@ -0,0 +1,361 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+from typing import List, Optional, Tuple, TypeVar, Union
+
+import nncf.experimental.tensor as tensor  # pylint: disable=consider-using-from-import
+from nncf.experimental.tensor.enums import TensorDataType
+from nncf.experimental.tensor.enums import TensorDeviceType
+
+TTensor = TypeVar("TTensor")
+
+
+@functools.singledispatch
+def device(a: TTensor) -> TensorDeviceType:
+    """
+    Return the device of the tensor.
+
+    :param a: The input tensor.
+    :return: The device of the tensor.
+    """
+    return device(a.data)
+
+
+@functools.singledispatch
+def squeeze(a: TTensor, axis: Optional[Union[int, Tuple[int]]] = None) -> TTensor:
+    """
+    Remove axes of length one from a.
+
+    :param a: The input tensor.
+    :param axis: Selects a subset of the entries of length one in the shape.
+    :return: The input array, but with all or a subset of the dimensions of length 1 removed.
+      This is always a itself or a view into a. Note that if all axes are squeezed,
+      the result is a 0d array and not a scalar.
+    """
+    if isinstance(a, tensor.Tensor):
+        return tensor.Tensor(squeeze(a.data, axis=axis))
+    return NotImplemented(f"Function `squeeze` is not implemented for {type(a)}")
+
+
+@functools.singledispatch
+def flatten(a: TTensor) -> TTensor:
+    """
+    Return a copy of the tensor collapsed into one dimension.
+
+    :param a: The input tensor.
+    :return: A copy of the input tensor, flattened to one dimension.
+    """
+    if isinstance(a, tensor.Tensor):
+        return tensor.Tensor(flatten(a.data))
+    return NotImplemented(f"Function `flatten` is not implemented for {type(a)}")
+
+
+@functools.singledispatch
+def max(a: TTensor, axis: Optional[Union[int, Tuple[int]]] = None) -> TTensor:  # pylint: disable=redefined-builtin
+    """
+    Return the maximum of an array or maximum along an axis.
+
+    :param a: The input tensor.
+    :param axis: Axis or axes along which to operate. By default, flattened input is used.
```

Comment:
Isn't it more general to call the parameter `axes` then and accept only list/tuple?